app/lib/features/attachments/widgets/reference_attachment_item.dart

```diff
@@ -1,213 +1,57 @@
+import 'package:acter/features/tasks/widgets/task_list_item_card.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' show Attachment;
 import 'package:flutter/material.dart';
-import 'package:flutter_easyloading/flutter_easyloading.dart';
-import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
-import 'package:go_router/go_router.dart';
-import 'package:phosphor_flutter/phosphor_flutter.dart';
 
 // Attachment item UI for References
 class ReferenceAttachmentItem extends ConsumerWidget {
   final Attachment attachment;
-  final RefDetails refDetails;
   final bool canEdit;
 
-  // whether item can be viewed on gesture
-  final bool? openView;
-
   const ReferenceAttachmentItem({
     super.key,
     required this.attachment,
-    required this.refDetails,
     this.canEdit = false,
-    this.openView,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    return Container(
-      decoration: BoxDecoration(
-        color: Theme.of(context).colorScheme.surface,
-        borderRadius: BorderRadius.circular(8),
-        border: Border.all(color: Theme.of(context).unselectedWidgetColor),
-      ),
-      child: ListTile(
-        onTap: () => onTapRefAttachment(context, ref),
-        leading: refAttachmentIcons(ref),
-        title: refAttachmentTitle(context, ref),
-        subtitle: refAttachmentSubTitle(),
-        trailing: actionMenu(context),
-      ),
-    );
-  }
+    final defaultWidget = SizedBox.shrink();
 
-  Widget refAttachmentTitle(BuildContext context, WidgetRef ref) {
-    final defaultTitle = Text(refDetails.title() ?? L10n.of(context).unknown);
-    final refObjectType = refDetails.typeStr();
-    final refObjectId = refDetails.targetIdStr();
+    final refDetails = attachment.refDetails();
+    if (refDetails == null) return defaultWidget;
 
-    if (refObjectId == null) return defaultTitle;
-    switch (refObjectType) {
-      case 'pin':
-        final pin = ref.watch(pinProvider(refObjectId)).valueOrNull;
-        return pin == null ? defaultTitle : Text(pin.title().toString());
-      case 'calendar-event':
-        final event = ref.watch(calendarEventProvider(refObjectId)).valueOrNull;
-        return event == null ? defaultTitle : Text(event.title().toString());
-      case 'task-list':
-        final taskList = ref.watch(taskListProvider(refObjectId)).valueOrNull;
-        return taskList == null
-            ? defaultTitle
-            : Text(taskList.name().toString());
-      default:
-        return defaultTitle;
-    }
-  }
-
-  Widget refAttachmentSubTitle() {
-    return Text(refDetails.typeStr().toString());
-  }
-
-  Widget refAttachmentIcons(WidgetRef ref) {
-    final defaultIcon = PhosphorIcon(PhosphorIconsThin.tagChevron, size: 30);
-    final refObjectType = refDetails.typeStr();
     final refObjectId = refDetails.targetIdStr();
+    final refObjectType = refDetails.typeStr();
```

Comment:
I thought we had agreed to make this a re-useable external widget that the attachment would just pull in? This will become a bit more messy when we need to add support for joining spaces, etc...

Reply:
Sorry didn't get you in this point, can you explain bit more hear?